doc/how-to-integrate-into-nix.rst

```diff
@@ -1,5 +1,5 @@
-Nix Integration
-===============
+How to integrate Cabal into Nix
```

Comment:
Do we need this? IIRC there's a ticket noting that the functionality is broken and suggesting to delete it.

Reply:
I kept it for now to show how a consistent guide naming could look like.

Reply:
There is already an open PR for the deletion of this section (which also removes the nix integration from cabal itself) here: https://github.com/haskell/cabal/pull/9239

Reply:
Yes, I think, it's better to remove it ASAP (at least the docs): it's a wart that is actively harming ux.

Reply:
Done